plugins/woocommerce-blocks/assets/js/blocks/product-filters/inner-blocks/price-filter/frontend.ts

```diff
@@ -2,232 +2,102 @@
-		searchParams.delete( 'max_price' );
-	}
-
-	searchParams.forEach( ( _, key ) => {
-		if ( /query-[0-9]+-page/.test( key ) ) searchParams.delete( key );
-	} );
-
-	return url.href;
+type PriceFilterContext = {
+	minPrice: number;
+	maxPrice: number;
+	minRange: number;
+	maxRange: number;
 };
 
-const debounceUpdate = debounce( ( context, element, event ) => {
-	const { decimalSeparator } = getCurrency();
-	const { minRange, minPrice, maxPrice, maxRange } = context;
-	const type = event.target.name;
-	const value = parseInt(
-		event.target.value
-			.replace( new RegExp( `[^0-9\\${ decimalSeparator }]+`, 'g' ), '' )
-			.replace( new RegExp( `\\${ decimalSeparator }`, 'g' ), '.' ),
-		10
-	);
-
-	const currentMinPrice =
-		type === 'min'
-			? Math.min( Number.isNaN( value ) ? minRange : value, maxPrice )
-			: minPrice;
-
-	const currentMaxPrice =
-		type === 'max'
-			? Math.max( Number.isNaN( value ) ? maxRange : value, minPrice )
-			: maxPrice;
-
-	if ( type === 'min' ) {
-		element.ref.value = currentMinPrice;
-	} else if ( type === 'max' ) {
-		element.ref.value = currentMaxPrice;
-	}
-
-	context.minPrice = currentMinPrice;
-	context.maxPrice = currentMaxPrice;
-
-	navigate(
-		getUrl( {
-			minRange,
-			maxRange,
-			minPrice: currentMinPrice,
-			maxPrice: currentMaxPrice,
-		} )
-	);
-}, 1000 );
-
-const constrainRangeSliderValues = (
-	/**
-	 * Tuple containing min and max values.
-	 */
-	values: [ number, number ],
-	/**
-	 * Min allowed value for the sliders.
-	 */
-	min?: number | null,
-	/**
-	 * Max allowed value for the sliders.
-	 */
-	max?: number | null,
-	/**
-	 * Step value for the sliders.
-	 */
-	step = 1,
-	/**
-	 * Whether we're currently interacting with the min range slider or not, so we update the correct values.
-	 */
-	isMin = false
-): { minPrice: number; maxPrice: number } => {
-	let [ minValue, maxValue ] = values;
-
-	const isFinite = ( n: number | null | undefined ): n is number =>
-		Number.isFinite( n );
-
-	if ( ! isFinite( minValue ) ) {
-		minValue = min || 0;
-	}
-
-	if ( ! isFinite( maxValue ) ) {
-		maxValue = max || step;
-	}
-
-	if ( isFinite( min ) && min > minValue ) {
-		minValue = min;
-	}
-
-	if ( isFinite( max ) && max <= minValue ) {
-		minValue = max - step;
-	}
-
-	if ( isFinite( min ) && min >= maxValue ) {
-		maxValue = min + step;
-	}
-
-	if ( isFinite( max ) && max < maxValue ) {
-		maxValue = max;
-	}
-
-	if ( ! isMin && minValue >= maxValue ) {
-		minValue = maxValue - step;
-	}
-
-	if ( isMin && maxValue <= minValue ) {
-		maxValue = minValue + step;
-	}
-
-	return { minPrice: minValue, maxPrice: maxValue };
-};
+const debounceUpdate = debounce(
+	(
+		context: PriceFilterContext,
+		parentContext: ProductFiltersContext,
+		price: Record< string, string >
+	) => {
+		Object.keys( price ).forEach( ( key ) => {
+			if ( key !== 'min_price' && key !== 'max_price' ) {
+				delete price[ key ];
+				return;
+			}
+			if ( ! price[ key ] ) {
+				delete price[ key ];
+			}
+		} );
```

Comment:
I believe we should avoid mutating the "price" object by directly deleting its properties, so we avoid violating the immutability principle.

I'd recommend to use `reduce` method here. This way we can keep only valid keys with non-empty values. It would be something like this:

```suggestion
		const priceWithValidKeys = Object.keys(price).reduce((acc, key) => {
		if ((key === 'min_price' || key === 'max_price') && price[key]) {
			acc[key] = price[key]; 
		}
		return acc;
	}, {} as Record<string, string>);
```

Reply:
Good catch here too, thanks!